packages/react-compose/src/resolveSlotProps.ts

```diff
@@ -0,0 +1,51 @@
+import * as React from 'react';
+import { OptionsResolverResult, defaultMappedProps } from './createOptionsResolver';
+
+export const NullRender = () => null;
+
+/**
+ * Helper utility which takes in a classes array from compose options, resolves functions,
+ * merges them into a final result, and distributes classnames to slotProps within the given
+ * resolver result object.
+ */
+export function resolveSlotProps(result: OptionsResolverResult): OptionsResolverResult {
+  const { state, slots, slotProps } = result;
+
+  Object.keys(slots).forEach((slotName: string) => {
+    const slot = slots[slotName];
+    let slotProp = state[slotName];
+
+    if (slot && slotProp !== undefined && slotProp !== null) {
+      const slotPropType = typeof slotProp;
+      const isLiteral = slotPropType === 'string' || slotPropType === 'number' || slotPropType === 'boolean';
+
+      // If the slot prop is a literal or JSX, pass it as a child of the slot.
+      if (isLiteral || React.isValidElement(slotProp)) {
+        const mappedProp =
+          (slot && slot.shorthandConfig && slot.shorthandConfig.mappedProp) || defaultMappedProps[slot] || 'children';
+        slotProp = { [mappedProp]: slotProp };
+      }
+
+      // If the children is a function, replace the slot.
+      if (typeof slotProp.children === 'function') {
```

Comment:
Is this going to be documented somewhere? This is interesting hidden behavior. Especially since the slot that I provide could supply its own data stuff down to the children.

Reply:
Yep. This is documented in v0 shorthand documentation, so I'm trying to preserve the logic to converge slot approaches.